nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -0,0 +1,404 @@
+                        # Spectrogram to Mel Spectrogram
+                        spec = dali.fn.mel_filter_bank(
+                            spec, sample_rate=self.sample_rate, nfilter=self.n_mels, normalize=True,
+                            freq_low=self.freq_low, freq_high=self.freq_high
+                        )
+                        # Mel Spectrogram to MFCC
+                        if feature_type == 'mfcc':
+                            spec = dali.fn.mfcc(spec, n_mfcc=self.n_mfcc)
+
+                    # Logarithm
+                    if self.log_zero_guard_type == 'add':
+                        spec = spec + self.log_zero_guard_value
+
+                    spec = dali.fn.to_decibels(
+                        spec, multiplier=math.log(10), reference=1.0, cutoff_db=math.log(self.log_zero_guard_value)
+                    )
+
+                    # Normalization
+                    spec = dali.fn.normalize(spec, axes=self.normalization_axes)
+
+                    # Extracting the length of the spectrogram
+                    shape_start = dali.types.Constant(np.array([1], dtype=np.float32), device='cpu')
+                    shape_len = dali.types.Constant(np.array([1], dtype=np.float32), device='cpu')
+                    spec_len = dali.fn.slice(
+                        dali.fn.shapes(spec), shape_start, shape_len, normalized_anchor=False, normalized_shape=False, axes=(0,)
+                    )
+
+                    # Pads feature dimension to be a multiple of `pad_to` and the temporal dimension to be as big as the largest sample (shape -1)
+                    spec = dali.fn.pad(
+                        spec, fill_value=self.pad_value, axes=(0, 1), align=(self.pad_to, 1), shape=(1, -1)
+                    )
+
+                    pipe.set_outputs(spec, spec_len, transcript, transcript_len)
+
+            # Building DALI pipeline
+            pipe.build()
+
+        if has_preprocessor:
+            output_names = ['processed_signal', 'processed_signal_len', 'transcript_raw', 'transcript_raw_len']
+        else:
+            output_names = ['audio', 'audio_len', 'transcript_raw', 'transcript_raw_len']
+
+        self._iter = DALIPytorchIterator(
+            self.pipes, output_map=output_names, reader_name="Reader",
+            fill_last_batch=True, dynamic_shape=True, auto_reset=True
+        )
+
+        # TODO come up with a better solution
+        class DummyDataset:
+            def __init__(self, parent):
+                self.parent = parent
+            def __len__(self):
+                return self.parent.size
+        self.dataset = DummyDataset(self)  # Used by NeMo
+
+    def reset(self):
+        self._iter.reset()
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        return self.__next__()
+
+    @property
+    def size(self):
+        return self._iter.size
+
+    def __len__(self):
+        return math.ceil(self._iter.size / self._iter.batch_size)
```

Comment:
The most recent DALI FW plugin already supports __len__, please use that instead:
```suggestion
        return len(self._iter)
```